### PR TITLE
[DRAFT] Initial spike at switching over to typescript

### PR DESCRIPTION
**Pull request by mjgiarlo**
This is admittedly a complete shot in the dark based on what documentation I found by googling, and is both incomplete and (most likely) broken at the moment. But perhaps it gets us closed to the goal of moving to TypeScript.

## Why was this change made?

Connects to #16 

## How was this change tested?

It wasn't.

## Which documentation and/or configurations were updated?

JS/webpacker stuff.

### CODE REVIEW

File: app/javascript/packs/application.ts [hunk 1]
```diff
@@ -20,4 +25,4 @@ import 'stylesheets/main'
 // const images = require.context('../images', true)
 // const imagePath = (name) => images(name, true)
 
-require('modules/validate-forms')
+import * from './modules/validate-forms'
```
Comment (mjgiarlo) [inline app/javascript/packs/application.ts:28]: My IDE complains about this. Not sure if that means I need to tweak the TS/JS support in the IDE or if it's just plain wrong. Have not tested in browser yet.

File: app/javascript/modules/validate-forms.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
   'use strict'
   window.addEventListener('turbolinks:load', function() {
     // Fetch all the forms we want to apply custom Bootstrap validation styles to
-    var forms = document.querySelectorAll('.needs-validation')
+    const forms = document.querySelectorAll('.needs-validation')
```
Comment (mjgiarlo) [inline app/javascript/modules/validate-forms.ts:6]: Done to appease eslint.

File: package.json [hunk 3]
```diff
@@ -2,17 +2,27 @@
...
-    "@rails/ujs": "^6.0.0",
+    "@rails/activestorage": "^6.0.3-4",
+    "@rails/ujs": "^6.0.3-4",
     "@rails/webpacker": "5.2.1",
     "bootstrap": "^5.0.0-alpha1",
     "dropzone": "^5.7.2",
     "stimulus": "^1.1.1",
-    "turbolinks": "^5.2.0"
+    "turbolinks": "^5.2.0",
+    "typescript": "^4.0.3"
```
Comment (jcoyne) [inline package.json:14]: Should typescript be a devDependency?

File: package.json [hunk 4]
```diff
@@ -2,17 +2,27 @@
   "name": "happy_heron",
   "private": true,
   "dependencies": {
+    "@babel/preset-typescript": "^7.12.0",
```
Comment (jcoyne) [inline package.json:5]: devDependency?

File: package.json (hunk 3 above)
Comment (mjgiarlo) [reply package.json:14]: In short, I don't know. This process was not terribly well documented, and I followed the guides I found, which led to this. I leave this PR here for someone else to take forward, or not.

Comment (jcoyne) [reply package.json:14]: Hmm, maybe it doesn't matter: https://github.com/facebook/create-react-app/issues/6180#issuecomment-453640473

**Comment by mjgiarlo** [comment]
@jcoyne given this is busted, I wonder if it's worth keeping it around. I might be inclined to nuke it. thoughts?

**Comment by jcoyne** [comment]
🤷

**Pull request closed without merging**